models/cartDao.js

```diff
@@ -77,9 +78,35 @@ const getProductById = async (productId) => {
   return product;
 };
 
+const getCarts = async (userId) => {
+  const [cartItems] = await Promise.all([
+    dataSource.query(
+      `SELECT 
+        carts.product_id,
+        products.id,
+        products.price,
+        products.weight,
+        carts.quantity 
+      FROM 
+        carts
+      JOIN
+        products
+      ON
+        carts.product_id = products.id
+      WHERE 
+        user_id = ?`,
+      [userId]
+    ),
+  ]);
```

Comment:
1. Promise.all을 사용하는 상황이 압니다. 제거해 주세요.
2. cartItem을 불러오는 곳에서 []로 감싸게 되면 여러 데이터 중 1개만 가져오게 됩니다. 

```suggestion
const getCarts = async (userId) => {
  const cartIemts = await dataSource.query(
      `SELECT 
        carts.product_id,
        products.id,
        products.price,
        products.weight,
        carts.quantity 
      FROM 
        carts
      JOIN
        products
      ON
        carts.product_id = products.id
      WHERE 
        user_id = ?`,
      [userId]
    );
```